onnxruntime/contrib_ops/cuda/bert/bert_padding.cu

```diff
@@ -361,6 +362,55 @@ void LaunchTrtSequenceOffset(int* trt_mha_padding_offset,
   }
 }
 
+// only support pure left padding with mask 0 on leading left,
+// or pure right padding with mask 0 on tailing right of each masks vector
+// for batch.
+__global__ void __launch_bounds__(kMAX_THREADS_PER_BLOCK)
+    getTrtSequenceOffset2d(int* trt_mha_padding_offset,
+                           const int* attention_masks,
+                           const int batch_size,
+                           const int sequence_length) {
+
+    typedef cub::BlockReduce<int, kMAX_THREADS_PER_BLOCK> BlockReduce;
+    __shared__ typename BlockReduce::TempStorage temp_storage;
+
+    const int batch_id = blockIdx.x;
+    const int* batch_mask = attention_masks + (batch_id * sequence_length);
+    const bool counting_on_zero = (batch_mask[0] == 0);
+    int biggest_position = 0;
+
+    for (int i = threadIdx.x; i < sequence_length; i += blockDim.x) {
+      if (counting_on_zero == (batch_mask[i] == 0)) {
+        biggest_position = i;
+      }
+    }
+
+    int leading_count = BlockReduce(temp_storage).Reduce(biggest_position, cub::Max(), blockDim.x);
+
+    if (threadIdx.x == 0) {
+      if (!counting_on_zero) { // default right side padding
+        trt_mha_padding_offset[2 * batch_id] = (batch_id * sequence_length);
+        trt_mha_padding_offset[2 * batch_id + 1] = (batch_id * sequence_length) + leading_count + 1;
+      } else { // left side padding
+        trt_mha_padding_offset[2 * batch_id] = (batch_id * sequence_length) + leading_count + 1;
```

Comment:
Could you add a unit test for left side padding and test it enable & disable fused causal?

Reply:
Let's use different checkin for the test cases.